### DISCUSSION TITLE
GC9A01 - Waveshare display Issue Coding Images

### DISCUSSION THREAD

**Discussion by Box-O-Rocks** [micropython/micropython#10582] [RP2040 / Pico]
Would anyone have experience with coding a waveshare display?  I bought a 1.28 round display which requires the GC9A01 module but I can't get much information information for coding the display in micropython.  The source code I'm using for the display came from [url]https://github.com/russhughes/gc9a01_mpy/blob/main/firmware/RP2W/firmware.uf2[/url] , and I found a bit map program from [url]https://www.youtube.com/watch?v=a7MzPA0T_MM[/url]... but I'm not sure how to properly write my desired display into the code.  

I can't imagine it is difficult.  I can get text to display... well after reworking some examples  🙄  I'm still a beginner at coding. 

Thank you for any help.

**Comment by jimmo** [comment]
hi @Box-O-Rocks -- the library you're using isn't part of this repo but there will almost certainly be people here that can help.

Have you seen the examples in https://github.com/russhughes/gc9a01_mpy/tree/main/examples and the documentation at https://github.com/russhughes/gc9a01_mpy/blob/main/README.md

What would you like your script to do? Perhaps you can share what you have already.

**Reply by Box-O-Rocks** [reply]
Oops, thanks @jimmo.  Should I move the post somewhere else? 

Honestly, all I am looking for the script to do is take an image whether it be calling the jpg file, or the bitmap .py file and outputting on the display.  I've seen it done plenty of times on arduino, but I would like to try it in micropython.  

Thank you for the threads, I will check them out and see if they help.

**Reply by Box-O-Rocks** [reply]
Well, I feel dumb.  I knew it was supposed to be easy.  I was executing the commands incorrectly.  I got my display to appear, and now that I know how the commands work, I can be more creative. 

Thank you again @jimmo
